src/Experimental/RAGTools/preparation.jl

```diff
@@ -719,31 +720,31 @@ function build_index(
     return index
 end
 
-using Pinecone: Pinecone, init_v3, Index
+using Pinecone: Pinecone, PineconeContextv3, PineconeIndexv3, init_v3, Index
+# TODO: change docs
 """
     build_index(
-        indexer::PTPineconeIndexer;
+        indexer::PineconeIndexer;
         namespace::AbstractString,
-        schema::AbstractPromptSchema = OpenAISchema();
         verbose::Integer = 1,
         index_id = gensym("PTPineconeIndex"),
         cost_tracker = Threads.Atomic{Float64}(0.0))
 
-Builds a `PTPineconeIndex` containing a Pinecone context (API key, index and namespace).
+Builds a `PineconeIndex` containing a Pinecone context (API key, index and namespace).
 """
 function build_index(
-        indexer::PTPineconeIndexer,
+        indexer::PineconeIndexer,
+        context::Pinecone.PineconeContextv3 = Pinecone.init_v3(""),
+        index::Pinecone.PineconeIndexv3 = "",
```

Comment:
this argument is confusing. is it the namespace/string?

Reply:
It is the name of the [Pinecone index](https://docs.pinecone.io/guides/indexes/understanding-indexes). Maybe calling it `pinecone_index` (and `pinecone_context` for the one above) would be better? And I would add details in the docs.